torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1337,16 +1337,18 @@ void ProcessGroupNCCL::heartbeatMonitor() {
     cpp_dumper.value()([](const std::string& line) { LOG(INFO) << line; });
   }
 
-  // Store debug info to storage if no other thread does it. (By default to
-  // local disk)
-  std::future<bool> asyncDebugDump = std::async(
-      std::launch::async, [this]() { return this->dumpDebuggingInfo(); });
-
-  // wait for the dump until timeout
-  waitForFutureOrTimeout(
-      asyncDebugDump,
-      std::chrono::milliseconds(waitTimeoutDumpInMilSec_),
-      "Flight recorder dump in heartbeatMonitor");
+  if (checkDumpSignal && shouldDump_.load()) {
+    // Store debug info to storage if no other thread does it. (By default to
+    // local disk)
+    std::future<bool> asyncDebugDump = std::async(
+        std::launch::async, [this]() { return this->dumpDebuggingInfo(); });
+
+    // wait for the dump until timeout
+    waitForFutureOrTimeout(
+        asyncDebugDump,
+        std::chrono::milliseconds(waitTimeoutDumpInMilSec_),
+        "Flight recorder dump in heartbeatMonitor");
+  }
```

Comment:
Do you want add something like:
``` 
else if (shouldDump_.load) {
      LOG(INFO) << "Not dumping coz I'm not 0";
 ```
 or will it be too chatty? or the log won't be useful?

Reply:
Yes, Probably a bit chatty as we know all non 0 thread should not dump. But this reminds me we should add some log when we first detected something wrong